pkgs/dart_mcp/lib/src/server/resources_support.dart

```diff
@@ -65,9 +76,26 @@ base mixin ResourcesSupport on MCPServer {
     (result.capabilities.resources ??= Resources())
       ..listChanged = true
       ..subscribe = true;
+    _resourceListChangedController.stream
+        .throttle(resourceUpdateThrottleDelay, trailing: true)
+        .listen(
+          (_) => sendNotification(
+            ResourceListChangedNotification.methodName,
+            ResourceListChangedNotification(),
+          ),
+        );
     return result;
   }
 
+  @override
+  Future<void> shutdown() async {
+    await super.shutdown();
+    await _resourceListChangedController.close();
+    final subscribed = _subscribedResources.values.toList();
+    _subscribedResources.clear();
+    await Future.wait([for (var controller in subscribed) controller.close()]);
```

Comment:
[optional] It's little shorter to use use the `Iterable<Future>.wait` extension.

```suggestion
    await subscribed.map((c) => c.close()).wait;
```